### ISSUE TITLE
Build failure with rust nightly

### ISSUE THREAD

**Issue by chandraprakash** [maidsafe/crust#149]
rustc 1.2.0-nightly (2f5683913 2015-06-18)

**Comment by vinipsmaker** [comment]
This recent build failures comes from two sources:

- We should use `#![feature(arc_weak)]` now.
- [`scoped` was deprecated](https://github.com/rust-lang/rust/commit/d645f8fc286af1841f1a352ddc7282c6e9a7dd8f).
  - [The reason behind deprecation is old](https://github.com/rust-lang/rust/issues/24292).
  - [This post is a good reading on the problem](http://smallcultfollowing.com/babysteps/blog/2015/04/29/on-reference-counting-and-leaks/).
  - [The discussion still heated](https://github.com/rust-lang/rfcs/pull/1084) and I'm not sure they'll provide a replacement soon.
    - So, I'm going to replace all `scoped` usages with `spawn`.

**Comment by chandraprakash** [comment]
+1 .. Thanks Vinícius

**Comment by vinipsmaker** [comment]
Previously, [`join` couldn't fail](https://doc.rust-lang.org/nightly/std/thread/struct.JoinGuard.html#method.join), but outside of the `scoped` API, [this scenario might happen](https://doc.rust-lang.org/nightly/std/thread/struct.JoinHandle.html#method.join).

I couldn't easily convert `Err(Box<Any + Send + 'static>)` to `Err(std::io::Error)`, so I used `unwrap` for now. Possibly I'll refactor to use some adapter (e.g. `Err(Error::new(Other, /* ... */))` or change `BroadcastAcceptor::accept` return type. These solutions touch public interface and I would need some time to make sure I don't break anything, but I was told you guys are in a hurry to getting this thing building for now, so I used `unwrap`.
